server/src/main/java/org/elasticsearch/script/ScriptHeuristicScript.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import java.util.Map;
+
+/**
+ * A script used in script heuristics.
+ */
+public abstract class ScriptHeuristicScript {
```

Comment:
This name is confusing having "script" twice. Can we call it `HeuristicScoreScript`? Or better yet, something specific to the agg it is used in (significant terms).

Reply:
`SignificantTermsHeuristicScoreScript` or too long? :)

Reply:
It's long, but probably better in this case too nur have any possible confusion with regular scoring scripts.